public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -494,6 +494,11 @@ export class DashboardModel implements TimeModel {
     for (const panel of this.panels) {
       panel.configRev = 0;
     }
+
+    if (this.panelInEdit) {
+      this.panelInEdit.hasSavedPanelEditChange = this.panelInEdit.configRev > 0;
```

Comment:
Trying to understand this - the `configRev` changes with any change provided via panel editor. But it doesn't mean the model was changed. So now, when we go back to dashboard, we reset the `configRev` so that we perceive a panel as **not changed** ?

Reply:
@dprokop 
if configRev is > 0 that means there has been a config change. Since we are clearing here (as this is called from save dashboard), we need to remember that we saved a change, and hence configRev is 0 but there is a change there that is still not reflected in the panel in the dashboard (only in panelInEdit)

Reply:
Ah, this now makes sense, thanks!